botocore/httpchecksum.py

```diff
@@ -387,6 +399,19 @@ def _apply_request_header_checksum(request):
     digest = checksum_cls().handle(request["body"])
     request["headers"][location_name] = digest
 
+    checksum_algorithm_name = algorithm["algorithm"]
+    request_checksum_calculation = request["context"][
+        "client_config"
+    ].request_checksum_calculation
+    request_checksum_calculation_feature_id = (
+        "FLEXIBLE_CHECKSUMS_REQ_" + request_checksum_calculation.upper()
+    )
```

Comment:
As mentioned above, I think this check specifically should go in `args.py` if possible. That's where we determine what the config values are.